### PR TITLE
DRILL-8106: do not use cellIterator because it skips empty cells

### PR DESCRIPTION
**Pull request by pjfanning**
# [DRILL-8106](https://issues.apache.org/jira/browse/DRILL-DRILL-8106): do not use cellIterator because it skips empty cells

## Description

@cgivre sent me an excel file that has missing cells and this throws off the existing code.

## Documentation
None expected

## Testing
Unit test - yet to be written

### CODE REVIEW

**Comment by pjfanning** [comment]
Current PR does not yet fully fix the code - I'm still getting issues with the sample file - but a different one from when I started.

**Comment by pjfanning** [comment]
I think the code is now basically fixed.

**Review by cgivre — COMMENTED** [review]
@pjfanning 
This looks good to me.  I had one question about using a `short` vs `int`, but other than that, LGTM +1.  Thanks as always for the quick fix!

File: contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java [hunk 1]
```diff
@@ -312,41 +311,31 @@ private void getColumnHeaders(SchemaBuilder builder) {
     //If there are no headers, create columns names of field_n
     if (readerConfig.headerRow == -1) {
       String missingFieldName;
-      int i = 0;
 
-      for (Cell c : currentRow) {
-        missingFieldName = MISSING_FIELD_NAME_HEADER + (i + 1);
+      for (short colNum = 0; colNum < currentRow.getLastCellNum(); colNum++) {
```
[the remark this answers is not in the document]
Comment (pjfanning) [reply contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java:315]: getLastCellNum returns a short so matching the type to that - excel only allows 16,384 columns and Short.MAX_VALUE = 32767

**Comment by pjfanning** [comment]
the new test case is failing and I need to get back to it to debug it

**Comment by pjfanning** [comment]
@cgivre I have the unit test passing locally now so hopefully this should be ready to merge after the CI build

File: contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cgivre) [reply contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java:315]: Gotcha.

**Review by cgivre — APPROVED** [review]
Assuming CI passes, LGTM +1. Thanks @pjfanning for this!

**Pull request merged**